OSDOCS 6914: Updating pod examples to comply with restricted PSA (App…

bergerhoffer opened the pull request:
…lications book)

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-6914

Link to docs preview:
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-examples
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-configuring_application-health
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/quotas/quotas-setting-per-project.html#setting-resource-quota-for-extended-resources_quotas-setting-per-project

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

bergerhoffer commented:
@s-urbaniak @stlaz PTAL

s-urbaniak commented:
/lgtm

s-urbaniak commented on modules/application-health-about.adoc:
```diff
@@ -104,6 +112,10 @@ metadata:
   labels:
     test: health-check
   name: my-application
+  securityContext:
```
oh sec, this needs to move under `spec`

s-urbaniak commented:
/lgtm cancel

s-urbaniak commented:
until https://github.com/openshift/openshift-docs/pull/62470#discussion_r1266819211 is resolved

s-urbaniak commented:
/lgtm

bergerhoffer commented:
FYI @Preeticp of this PR to update all pod examples in the applications book to comply with pod security admission standards. This is part of a larger effort to update all pod examples in the repo. Let me know if you have any questions or concerns on this. Thanks!

bergerhoffer commented:
@xingxingxia @gkarager This PR is ready for QE review. Again, please let me know if we should request QEs from this area to do the verification.

Preview links of the affected files are in the description.

xingxingxia commented:
@bergerhoffer , @gkarager is PTO, I saw your Jira issue's tagging. I was busy in Auth / cert-manager two parts work. So kindly expect one or two days' waiting before I check, thanks! Will check later.

bergerhoffer commented:
Thanks @xingxingxia! No hurry at all, I'm just letting you know which PRs are actually ready for QE review when you all get a chance. Some of the other PRs in the Jira still have open questions in them that we need to resolve first. Thanks!

xingxingxia commented:
> Again, please let me know if we should request QEs from this area to do the verification.

@bergerhoffer from the perspective of adding the "securityContext" fields, it is LGTM. From the verification of user-end, yes, need related QEs to help verify. At least I tried one of the modules/application-health-about.adoc, it has issue:
```
[xxia@2023-07-20 17:34:05 CST my]$ oc create -f <(echo "
apiVersion: v1
kind: Pod
metadata:
  labels:
    test: health-check
  name: my-application
spec:
  securityContext:
    runAsNonRoot: true
    seccompProfile:
      type: RuntimeDefault
  containers:
  - name: goproxy-app
    args:
    image: registry.k8s.io/goproxy:0.1
    readinessProbe:
      exec:
        command:
        - cat
        - /tmp/healthy
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop: [ALL]
")
pod/my-application created

# "0/1" shows the container is not ready:
[xxia@2023-07-20 17:34:21 CST my]$ oc get po
NAME             READY   STATUS    RESTARTS   AGE
my-application   0/1     Running   0          2m19s

[xxia@2023-07-20 17:36:32 CST my]$ oc describe po my-application
Name:             my-application
Namespace:        test
...
Events:
  Type     Reason          Age    From               Message
  ----     ------          ----   ----               -------
...
  Warning  Unhealthy  4s (x8 over 74s)  kubelet  (combined from similar events): Readiness probe errored: rpc error: code = Unknown desc = command error: time="2023-07-20T09:36:43Z" level=error msg="exec failed: unable to start container process: exec: \"cat\": executable file not found in $PATH"
, stdout: , stderr: , exit code -1
```
@gkarager comes back :smile: today, please help handle new tagging when there are, thanks : )

bergerhoffer commented:
Hi @lyman9966, I'm looking for someone to help test/verify these updates.  This PR updates all pod examples in the applications book to comply with pod security admission standards. This is part of a larger effort to update all pod examples in the repo. It looks like at least one of them has an issue when being tested (see comment above). 

I'm not sure which area owns these docs, but I found your name in verifying an old PR in this area. But let me know if someone else should help with this instead. Thanks!

kalexand-rh commented:
The `branch/enterprise-4.15` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.14. And any PR going into `main` must also target the latest version branch (enterprise-4.15).

If the update in your PR does NOT apply to version 4.15 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main

lyman9966 commented:
@bergerhoffer I can't open the docs preview, it showed: Site Not Found (e.g. https://62470--docspreview.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-examples)

kalexand-rh commented:
The `branch/enterprise-4.16` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.15. And any PR going into `main` must also target the latest version branch (enterprise-4.16).

If the update in your PR does NOT apply to version 4.16 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

bergerhoffer commented:
/retest

bergerhoffer commented:
HI @lyman9966, sorry for the long delay. I've updated the preview links (we changed preview systems). All should work now. Let me know how these changes look, thanks!

bergerhoffer commented on modules/application-health-about.adoc:
```diff
@@ -86,6 +90,10 @@ spec:
         command: <5>
```
@lyman9966 Do you happen to know a different image we could use here instead?

lyman9966 replied:
Sorry, I don't have this kind of image on hand. And I think we can't simply replace it with any image which satisfied PSA, but it should also support the command of readinessProbe:
readinessProbe:
      exec:
        command:
        - cat
        - /tmp/healthy
